server/src/main/java/io/crate/expression/scalar/ArraySetFunction.java

```diff
@@ -111,4 +116,45 @@ private void setElement(List<Object> source, int index, Object value) {
             source.add(value);
         }
     }
+
+    public static class Allocator {
```

Comment:
I'd tend to keep this private within the `UpdateAnalyzer` as it's not really intended for general use but for a very specific use case of the update statement.